crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -87,7 +87,9 @@ pub trait Environment {
     fn name(&self) -> EnvironmentName;
 
     /// Delete the Environment
-    fn delete(self) -> Result<(), EnvironmentError2>;
+    fn delete(self) -> Result<(), EnvironmentError2>
+    where
+        Self: Sized;
```

Comment:
Would using `Box` be simpler? Allows dropping `ConcreteEnvironment`. Seemed to compile: https://github.com/flox/flox/commit/097330295b661d20af3d8c2680f5acdf602a3983

Reply:
Would we expect to ever need a `ConcreteEnvironment`?

Reply:
`delete(Box<Self>)` is a bit weird, as we cant use it on `*Environment` structs _without_  first `Box`'ing  the struct.
Also `SyncEnvironement` or similar traits for method subsets are not implemented for _all_ envs (`ConcreteEnv::to_sync() -> Option<Box<dyn SyncEnv>>` should be preferable over implementing SyncEnv for all environments and throwing errors).
See also delete_symlinks/gc which should be only relevant for ManagedEnvs right (possibly we want gcroots for Remote Envs but that's a different question to ask)